ergo-wallet/src/main/scala/org/ergoplatform/wallet/interpreter/ErgoInterpreter.scala

```diff
@@ -32,15 +32,17 @@ class ErgoInterpreter(params: ErgoLikeParameters)(implicit IR: IRContext)
     * @return whether the box is spent properly according to the storage fee rule
     */
   protected def checkExpiredBox(box: ErgoBox, output: ErgoBoxCandidate, currentHeight: Height): Boolean = {
-    val maxStorageFee = params.storageFeeFactor * box.bytes.length
+    val storageFee = params.storageFeeFactor * box.bytes.length
 
-    val storageFeeCovered = box.value - maxStorageFee <= 0
+    val storageFeeNotCovered = box.value - storageFee <= 0
     val correctCreationHeight = output.creationHeight == currentHeight
-    val correctOutValue = output.value >= box.value - maxStorageFee
+    val correctOutValue = output.value >= box.value - storageFee
+
+    // all the registers except of R0 (monetary value) and R3 (creation height and reference) must be preserved
     val correctRegisters = ErgoBox.allRegisters.tail
       .forall(rId => rId == ErgoBox.ReferenceRegId || box.get(rId) == output.get(rId))
```

Comment:
There are a number of issues with this code:
1) This is very cryptic implementation of register comparions
2) the `tail` call allocates a new array before calling forall.

I suggest to create `sameNonMandatoryRegisters` method and compare mandatory registers explicitly (since only two of them must be compared)

Reply:
@aslesarenko that's true but not related to the scope of the PR. Please make an issue.

Reply:
Why the checking of registers are related to expiry validation?

Reply:
Got it by reading the code again